src/bindings/c/include/openvino/c/deprecated.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @file openvino/c/deprecated.h
+ * Defines macros for API deprecation
+ **/
+#pragma once
+
+//
+// The OPENVINO_DEPRECATED macro can be used to deprecate a function declaration. For example:
+//
+//     OPENVINO_DEPRECATED("replace with groxify");
+//     void frobnicate()
+//
+// The macro will expand to a deprecation attribute supported by the compiler,
+// so any use of `frobnicate` will produce a compiler warning.
+//
+
+#if defined(_WIN32)
+#    define OPENVINO_DEPRECATED(msg) __declspec(deprecated(msg))
+#    if __cplusplus >= 201402L
+#        define OPENVINO_ENUM_DEPRECATED(msg) [[deprecated(msg)]]
+#    else
+#        define OPENVINO_ENUM_DEPRECATED(msg)
+#    endif
+#elif defined(__INTEL_COMPILER)
+#    define OPENVINO_DEPRECATED(msg)      __attribute__((deprecated(msg)))
+#    define OPENVINO_ENUM_DEPRECATED(msg) OPENVINO_DEPRECATED(msg)
+#elif defined(__GNUC__)
+#    define OPENVINO_DEPRECATED(msg) __attribute__((deprecated(msg)))
+#    if __GNUC__ < 6 && !defined(__clang__)
+#        define OPENVINO_ENUM_DEPRECATED(msg)
+#    else
+#        define OPENVINO_ENUM_DEPRECATED(msg) OPENVINO_DEPRECATED(msg)
+#    endif
+#else
+#    define OPENVINO_DEPRECATED(msg)
+#    define OPENVINO_ENUM_DEPRECATED(msg)
+#endif
+
+// Suppress warning "-Wdeprecated-declarations" / C4996
+#if defined(_MSC_VER)
+#    define OPENVINO_DO_PRAGMA(x) __pragma(x)
+#elif defined(__GNUC__)
+#    define OPENVINO_DO_PRAGMA(x) _Pragma(#    x)
+#else
+#    define OPENVINO_DO_PRAGMA(x)
+#endif
+
+#if defined(_MSC_VER) && !defined(__clang__)
+#    define OPENVINO_SUPPRESS_DEPRECATED_START \
```

Comment:
maybe we need to give different names? To prevent clashes in case of C and C++ headers are mixed in the same source files.